crashes/office-dashboard.js

```diff
@@ -0,0 +1,138 @@
+var gWidth;
+var gHeight;
+
+function graph_error(target, err) {
+  MG.data_graphic({
+    width: gWidth,
+    height: gHeight,
+    target: target,
+    chart_type: "missing-data",
+    error: err,
+    missing_text: "Failed to load data.",
+  });
+}
+
+document.addEventListener("DOMContentLoaded", function() {
+  var s = d3.select("#beta-graph");
+  gWidth = parseInt(s.style("width"));
+  gHeight = parseInt(s.style("height"));
+
+  get_beta_bydate().then(
+    function(data) {
+      setup_channel_graph(data, "beta");
+    },
+    graph_error.bind(undefined, "#beta-graph")
+  );
+  get_aurora_bydate().then(
+    function(data) {
+      setup_channel_graph(data, "aurora");
+    },
+    graph_error.bind(undefined, "#aurora-graph")
+  );
+  get_nightly_bydate().then(
+    function(data) {
+      setup_channel_graph(data, "nightly");
+    },
+    graph_error.bind(undefined, "#nightly-graph")
+  );
+  get_crashes(326, "d2f89374fbb282f44b299e1267f9c99771f59773").then(
+    function(data) {
+      setup_e10s_graph(data);
+    },
+    graph_error.bind(undefined, "#e10s-graph")
+  );
+});
+
+function setup_channel_graph(data, channel) {
+  var last_version = null;
+  var markers = [];
+
+  var rows = data.query_result.data.rows.filter(function(row) {
+    return row.activity_date.getTime() < Date.now() - MS_PER_DAY * 2;
+  });
+  rows.forEach(function(row) {
+    if (row.leading_version != last_version) {
+      last_version = row.leading_version;
+      markers.push({activity_date: row.activity_date, label: row.leading_version});
+    }
+  });
+
+  MG.data_graphic({
+    data: rows,
+    width: gWidth,
+    height: gHeight,
+    target: "#" + channel + "-graph",
+    x_accessor: 'activity_date',
+    y_accessor: ['app_crash_rate', 'plugin_crash_rate'],
+    legend: ["App crashes - main+content", "Plugin crashes - NPAPI+GMP"],
+    legend_target: "#legend",
+    // baselines: [{value: 10, label: "max app-crash rate"}, {value: 7.5, label: "max plugin-crash rate"}],
+    markers: markers,
+    interpolate: "linear",
+    area: false,
```

Comment:
Refactor a shared options object out (like margins and graphics display) and merge them with `Object.assign` to reduce repeated lines.